pennylane/devices/default_qubit.py

```diff
@@ -674,28 +693,38 @@ def setup_execution_config(
             if option not in updated_values["device_options"]:
                 updated_values["device_options"][option] = getattr(self, f"_{option}")
 
-        mcm_config = config.mcm_config
+        mcm_config = self._setup_mcm_config(config, circuit)
+
+        updated_values["mcm_config"] = mcm_config
+        return replace(config, **updated_values)
 
-        if mcm_config.mcm_method == "device":
-            mcm_config = replace(mcm_config, mcm_method="tree-traversal")
+    def _setup_mcm_config(self, config: ExecutionConfig, tape: QuantumScript) -> MCMConfig:
```

Comment:
Any reason we aren't just passing `config.mcm_config`
```suggestion
    def _setup_mcm_config(self, mcm_config: MCMConfig, tape: QuantumScript) -> MCMConfig:
```

Reply:
changing locally.